src/System.Private.Uri/src/System/UriBuilder.cs

```diff
@@ -114,19 +81,12 @@ string extraValue
             {
                 Extra = extraValue;
             }
-            catch (Exception exception)
+            catch (ArgumentException)
```

Comment:
I'll let @davidsh answer.  Given his previous comments, though, my suggestion would simply be to delete this whole commit and leave the PR focused just on adding the tests, which generally look good to me.